### ISSUE TITLE
allow prop `transitionName` to be type of `object`

### ISSUE THREAD

**Issue by goldensunliu** [marnusw/react-css-transition-replace#12]
On this example I get the following warning.
```jsx
        <ReactCSSTransitionReplace
          component='div'
          transitionName={{
            enter: PreviewEnter,
            leave: PreviewLeave
          }}
          transitionEnterTimeout={400}
          transitionLeaveTimeout={400}
        >
          { this.renderPreview() }
        </ReactCSSTransitionReplace>
```
```
Failed propType: Invalid prop `transitionName` of type `object` supplied to `ReactCSSTransitionReplace`, expected `string`
```

I am expecting an object is also okay, according to `ReactCSSTransitionGroup`
https://facebook.github.io/react/docs/animation.html#custom-classes

The functionality seems to be there, but the warning is a red herring

**Comment by BerkeleyTrue** [comment]
:+1:

**Comment by marnusw** [comment]
Resolved in `v1.1.1`.
